script.js

```diff
@@ -1,4 +1,261 @@
+ */
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+/**
+ * Shuffle elements in the cardDeck array. Return the shuffled deck.
+ */
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex to shuffle the next pair of cards
+    currentIndex += 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+/**
+ * Deal Cards to the two players (think about how it can be modified for x players - store players in an array then loop for array length of players. Also think about how one deal card function can serve all purposes - deal 2 cards when its first turn meaning card array length is = 0)
+ */
+var dealCards = function (hand1, hand2) {
+  // Check if first deal - can i run a function that takes 2 parameters on only 1 parameter - i want to have a universal deal function
+  if (hand1.length < 1 && hand2.length < 1) {
+    var cardsDrawn = 0;
+    while (cardsDrawn < 2) {
+      var firstUserCardDraw = deck.pop();
+      hand1.push(firstUserCardDraw);
+      var firstCompCardDraw = deck.pop();
+      hand2.push(firstCompCardDraw);
+      cardsDrawn = cardsDrawn + 1;
+    }
+  }
+  // do I need to return anything here? how do i return both hand1 and hand2 - confused about what return does
+};
+
+/**
+ * Calculate Hand Scores
+ */
+var calculateHandScore = function (hand) {
+  var handIndex = 0;
+  var handScore = 0;
+  while (handIndex < hand.length) {
+    var handScore = handScore + Number(hand[handIndex].rank);
+    handIndex = handIndex + 1;
+  }
+  //using some to trigger conditional ace value change - unsure if totally correct
+  if (hand.some((card) => card.name === "Ace")) {
+    while (handScore <= blackjack - 10) {
+      handScore = handScore + 10;
+    }
+  }
+  return handScore;
+};
+
+/**
+ * Wordify a hand - ideally remove the comma from the last card
+ */
+var expressHandInWords = function (hand) {
+  var handIndex = 0;
+  var wordifiedHand = "";
+  // can I use the same handIndex var here?
+  while (handIndex < hand.length) {
+    var wordifiedHand = wordifiedHand + `${hand[handIndex].name} of ${hand[handIndex].suit} ,`;
+    handIndex = handIndex + 1;
+  }
+  return wordifiedHand;
+};
+
+/**
+ * Output the results - do i need to put all the parameters in the function definition if the name of the vars are all the same as in main?
+ */
+var showTheResults = function (userScore, computerScore, userHand, computerHand) {
```

Comment:
You do not need to pass in `userScore`, `computerScore`, `userHand`, `computerHand` if they are the same variables originally created in [lines 179 to 182](https://github.com/rocketacademy/basics-blackjack/blob/43d6c7a965c66360721d356d06b0c9255998ac35/script.js#L179-L182).

These variables are global variables, so they should be accessible by the functions without the need for parameters. 

However, if these variables are created in a particular function, say in `main`, they will then be scoped in the `main` function. If they are scoped to the `main` function, other functions will not be able to access those variables without them having been passed as parameters.